Improve the kfctl_existing_arrikto doc

cspavlou opened the pull request:
Add instructions for installing MetalLB
Add `Troubleshooting` section
Add architecture figure
Change the kfctl_existing_arrikto description in the Overview page

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/website/1056)
<!-- Reviewable:end -->

cspavlou commented:
/ assign @yanniszark

cspavlou commented:
/assign @yanniszark

yanniszark commented:
/lgtm

@sarahmaddox it would be great if you could also take a look, your comments are always very helpful :smile:

yanniszark commented:
/ok-to-test

sarahmaddox commented:
/assign

sarahmaddox commented:
Preview:
* https://deploy-preview-1056--competent-brattain-de2d6d.netlify.com/docs/started/k8s/overview/
* https://deploy-preview-1056--competent-brattain-de2d6d.netlify.com/docs/started/k8s/kfctl-existing-arrikto/

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
```diff
@@ -1,22 +1,156 @@
 ### Prerequisites
-- Kubernetes Cluster with LoadBalancer support. Refer [Load Balancer](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/) guide.
 
+- Kubernetes Cluster with LoadBalancer support.
```
It looks a little confusing to have a single bulleted list item here, especially when the folded portion "MetalLB deployment" is expanded. 

Recommendation: Remove the bullet and have a full sentence instead:

```
You need a Kubernetes Cluster with LoadBalancer support.

If you don't have...
```

cspavlou replied:
thanks, will change that

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
```diff
@@ -1,22 +1,156 @@
+
+If you don't have support for LoadBalancer on your cluster, please
+follow the instructions below to deploy MetalLB in [Layer 2
+mode](https://metallb.universe.tf/configuration/#layer-2-configuration).
```
Recommendation:

```
follow the instructions below to deploy MetalLB in Layer 2
mode. (You can read more about Layer 2 mode in the
[MetalLB docs](https://metallb.universe.tf/configuration/#layer-2-configuration).)
```

The reason for this suggestion is that my first inclination was to click the link and look for installation instructions there. The word "below" is ambiguous, unless we make it clear that the link leads to extra info.

cspavlou replied:
Thank Sarah, this makes sense

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
```diff
@@ -1,22 +1,156 @@
+    {{< /highlight >}}
+
+
+#### Ensure that MetalLB works as expected (optional)
```
This heading stands out a bit too much, especially as there isn't a matching heading for the previous section of the folded text. Also, the heading appears in the mini table of contents on the right even when the folded section is not expanded.

Suggestion: Make this sentence bold text instead of a heading:

```
**Ensure that MetalLB works as expected (optional):**
```

And also add a matching  heading at the top of the folded section:

```
**Deploy MetalLB:***
```

cspavlou replied:
thanks

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
```diff
@@ -1,22 +1,156 @@
+   addresses `10.0.0.100-10.0.0.110`. *You must modify these addresses based on
+   your environment*.
+
+    {{< highlight bash >}}
```
The indentation of this code snippet isn't rendering a copy-pastable code snippet. I haven't used the `{{< highlight bash >}}` shortcode before (I usually use 3 ticks). I suspect you need to play around with the indentation to get the rendering right.

cspavlou replied:
It seems that a numbered list containing _```bash_ breaks the numbering. That's why I used an alternative here. But since the list ends with this step, I can use _```bash` an fix this. Thank you

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
```diff
@@ -1,22 +1,156 @@
+
+#### Ensure that MetalLB works as expected (optional)
+
+a) Create a dummy service:
```
It'd be nice if you could use a numbered list rather than hard-coded letters, but I suspect you did this to get round the problem with rendering code snippets? Having the code snippet start at column 5 should work, though perhaps the fact that this text is within a folded section may complicate things.

cspavlou replied:
I replaced `3 ticks + bash` with `3 ticks` and everything worked. I prefer the `3 ticks + bash` thing, but you are right about the numbered list. Thanks

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
````diff
@@ -1,22 +1,156 @@
+service/nginx created
+```
+
+b) Ensure that MetalLB has actually allocated an IP for the service:
````
s/IP/IP address

cspavlou replied:
done

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
````diff
@@ -1,22 +1,156 @@
+service/nginx created
+```
+
+b) Ensure that MetalLB has actually allocated an IP for the service:
````
Remove the word "actually" as it can be confusing, and it doesn't add any meaning.

cspavlou replied:
done

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
````diff
@@ -1,22 +1,156 @@
+pod/nginx created
+```
+
+e) Ensure that MetalLB has assigned a node to announce the allocated IP:
````
s/IP/IP address

cspavlou replied:
done

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
````diff
@@ -1,22 +1,156 @@
+{"caller":"main.go:246","event":"serviceAnnounced","ip":"10.0.0.101","msg":"service has IP, announcing","pool":"default","protocol":"layer2","service":"default/nginx","ts":"2019-08-09T15:14:02.433876894Z"}
+```
+
+g) Check that MetalLB responds to ARP requests for the allocated IP:
````
s/IP/IP address

cspavlou replied:
done

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
````diff
@@ -65,7 +199,7 @@ To get the Istio Gateway IP, run:
 ```
 
-Get the IP and open it in a browser: `https://<ip>/`.
+Get the IP and open it in a browser: `https://<LoadBalancerIP>/`.
````
s/IP/IP address

cspavlou replied:
Thanks Sarah, I changed to `<LoadBalancerIP address>`

@yanniszark what do you think?

cspavlou replied:
I found two more occurrences and amended accordingly

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
```diff
@@ -420,6 +554,81 @@ This section focuses on setting up Dex to authenticate with an existing LDAP dat
 
+### Troubleshooting
+
+If the Kubeflow Dashboard is not available at `https://<LoadBalancerIP>` ensure that:
```
s/Dashboard/dashboard

(Because capitalising things can be confusing - it makes readers think they should see the brand name "Kubeflow Dashboard" somewhere on the UI.")

I see we call it a "Dashboard" earlier on this page too. Please would you change any existing occurrences while you're here?

cspavlou replied:
I agree and changed two more occurrences

sarahmaddox commented on content/docs/started/k8s/kfctl-existing-arrikto.md:
```diff
@@ -33,7 +167,7 @@ Follow these steps to deploy Kubeflow:
 export PATH=$PATH:"<path to kfctl>"
 export KFAPP="<your choice of application directory name>"
-export CONFIG="https://raw.githubusercontent.com/kubeflow/kubeflow/master/bootstrap/config/kfctl_existing_arrikto.0.6.yaml"
+export CONFIG="https://raw.githubusercontent.com/kubeflow/kubeflow/master/bootstrap/config/kfctl_existing_arrikto.yaml"
```
The link to the YAML file gives a 404.

yanniszark replied:
The name of the file is changed in this PR: https://github.com/kubeflow/kubeflow/pull/3857
Once it goes through, this comment should be addressed.

cspavlou replied:
Thanks @yanniszark 
@sarahmaddox I should have mentioned this

sarahmaddox commented:
> The name of the file is changed in this PR: [kubeflow/kubeflow#3857](https://github.com/kubeflow/kubeflow/pull/3857)
> Once it goes through, this comment should be addressed.

Thanks @yanniszark I'll put a hold on this PR to prevent it being merged until https://github.com/kubeflow/kubeflow/pull/3857 is merged.

/hold

cspavlou commented:
@sarahmaddox thank you for your comments. I implemented the proposed changes and also introduced a numbered list at the troubleshooting section.

@yanniszark please take a look

sarahmaddox commented:
/lgtm
/approve

Thanks @cspavlou this looks great. There's just one change that may need making: When you changed all `IP` to `IP address`, it's possible that two occurrences of `IP` in a message should not have been changed. They now read as `IP address assigned by controller`, and I seem to remember they were `IP assigned by controller` before.

I've approved this PR. When you're ready, I or someone else can remove the hold by saying `/hold cancel`, and add an LGTM if necessary, and the merge will go through.

cspavlou commented:
> /lgtm
> /approve
> 
> Thanks @cspavlou this looks great. There's just one change that may need making: When you changed all `IP` to `IP address`, it's possible that two occurrences of `IP` in a message should not have been changed. They now read as `IP address assigned by controller`, and I seem to remember they were `IP assigned by controller` before.
> 
> I've approved this PR. When you're ready, I or someone else can remove the hold by saying `/hold cancel`, and add an LGTM if necessary, and the merge will go through.

Thank you @sarahmaddox. I checked the messages, they do read `IP address assigned by controller`, so there is no need for further changes.

cspavlou commented:
/retest

cspavlou commented:
/test all

cspavlou commented:
@sarahmaddox , https://github.com/kubeflow/kubeflow/pull/3857 was merged into master. I think we are good to go with this PR. I rebased to master because of https://github.com/kubeflow/website/pull/1051

@yanniszark what do you think?

yanniszark commented:
I'm good with the changes.
/lgtm

sarahmaddox commented:
/hold cancel

Thanks everyone!
